src/libaktualizr-posix/ipuptanesecondary.cc

```diff
@@ -52,6 +52,36 @@ Uptane::SecondaryInterface::Ptr IpUptaneSecondary::create(const std::string& add
   return std::make_shared<IpUptaneSecondary>(address, port, serial, hw_id, pub_key);
 }
 
+SecondaryInterface::Ptr IpUptaneSecondary::connectAndCheck(const std::string& address, unsigned short port,
+                                                           EcuSerial serial, HardwareIdentifier hw_id,
+                                                           PublicKey pub_key) {
+  // try to connect:
+  // - if it succeeds compare with what we expect
+  // - otherwise, keep using what we know
+  try {
+    auto sec = IpUptaneSecondary::connectAndCreate(address, port);
+    if (sec != nullptr) {
+      auto s = sec->getSerial();
+      if (s != serial) {
+        LOG_ERROR << "Mismatch between secondary serials " << s << " and " << serial;
+      }
+      auto h = sec->getHwId();
+      if (h != hw_id) {
+        LOG_ERROR << "Mismatch between hardware ids " << h << " and " << hw_id;
+      }
+      auto p = sec->getPublicKey();
+      if (p != pub_key) {
+        LOG_ERROR << "Mismatch between public keys " << p.Value() << " and " << pub_key.Value();
+      }
+      // note: we still trust what the device gave us for now
+      return sec;
+    }
+  } catch (std::exception& e) {
```

Comment:
I think, an error or debug message would be helpful here.

Reply:
Added in latest version